src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala

```diff
@@ -179,17 +189,35 @@ class ErgoNodeViewSynchronizer(networkControllerRef: ActorRef,
       // should not be here, and so ReserveCostValue should not be used
       log.warn("Cost is empty in processMempoolResult")
     }
+
+    //TODO peer should always be set?  Throw exception?
+    val peer = processingResult.transaction.source.get
```

Comment:
source can be empty if transaction is submitted via local API. In this case we need to skip txInfo update.